src/d3a/models/area/event_dispatcher.py

```diff
@@ -175,3 +181,19 @@ def create_area_agents(self, is_spot_market, market):
         if self.area.parent:
             # Add inter area appliance to report energy
             self.area.appliance = InterAreaAppliance(self.area.parent, self.area)
+
+    def _delete_past_agents(self, timeslot, area_agent_member):
+        if not ConstSettings.GeneralSettings.KEEP_PAST_MARKETS:
+            delete_agents = [pm for pm in area_agent_member.keys() if
+                             self.area.current_market and pm < self.area.current_market.time_slot]
+            for pm in delete_agents:
+                for i in area_agent_member[pm]:
```

Comment:
Can you please exchange `i` with `area_agent` and `j` with engine or something similar and more verbose. Thanks!

Reply:
will do, thanks!